### PR TITLE
[EWPP-844] EWPP-846: Create formatter for JavaScript asset media.

### PR DESCRIPTION
**Pull request by nagyad**

### CODE REVIEW

File: modules/oe_media_js_asset/tests/src/Kernel/JavaScriptAssetUrlFormatterTest.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      'id' => 'test_js_asset',
+      'label' => 'Test JS asset',
+      'source' => 'oe_media_js_asset',
+    ]);
+
+    $view_display = \Drupal::service('entity_display.repository')->getViewDisplay('media', $media_type->id());
+    $this->display = $view_display;
+    $source = $media_type->getSource();
+    $this->fieldName = $source->getConfiguration()['source_field'];
+    $this->mediaType = $media_type;
```
Comment (brummbar) [inline modules/oe_media_js_asset/tests/src/Kernel/JavaScriptAssetUrlFormatterTest.php:88]: This is not needed, we are hardcoding the id in line 79.

File: modules/oe_media_js_asset/tests/src/Kernel/JavaScriptAssetUrlFormatterTest.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    $this->installEntitySchema('user');
+
+    $media_type = $this->createMediaType('javascript_asset', [
+      'id' => 'test_js_asset',
+      'label' => 'Test JS asset',
+      'source' => 'oe_media_js_asset',
+    ]);
+
+    $view_display = \Drupal::service('entity_display.repository')->getViewDisplay('media', $media_type->id());
+    $this->display = $view_display;
```
Comment (brummbar) [inline modules/oe_media_js_asset/tests/src/Kernel/JavaScriptAssetUrlFormatterTest.php:85]: Also this is not needed, as written below the media id is passed to the method so we can just load the display in the (only) test method below.

File: modules/oe_media_js_asset/tests/src/Kernel/JavaScriptAssetUrlFormatterTest.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      'bundle' => $this->mediaType->id(),
+      'name' => 'Test JS asset media',
+      'status' => TRUE,
+    ]);
+    $entity->{$this->fieldName}->environment = 'acceptance';
+    $entity->{$this->fieldName}->path = '/somejavascript.js';
+
+    // Verify that the oe_media_js_asset_url format has been applied.
+    $this->renderEntityFields($entity, $this->display);
+    $this->assertRaw('<script type="application/json" src="https://acceptance.europa.eu/webassets/somejavascript.js"></script>');
```
Comment (brummbar) [inline modules/oe_media_js_asset/tests/src/Kernel/JavaScriptAssetUrlFormatterTest.php:114]: We need to add coverage for:

- rendering of multiple js assets;
- rendering a non-existing environment configured.

**Review by brummbar — CHANGES_REQUESTED** [review]

**Review by brummbar — APPROVED** [review]
I pushed some commits to fix the tests and minor code readability.

**Review by brummbar — APPROVED** [review]

File: modules/oe_media_js_asset/src/Plugin/Field/FieldFormatter/JavaScriptAssetUrlFormatter.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    $environments = $config->get('environments');
+
+    foreach ($items as $delta => $item) {
+      if (!empty($environments[$item->environment]['url'])) {
+        $elements[$delta] = [
+          '#type' => 'html_tag',
+          '#tag' => 'script',
+          '#value' => '',
+          '#attributes' => [
+            'type' => 'application/javascript',
```
Comment (sergepavle) [inline modules/oe_media_js_asset/src/Plugin/Field/FieldFormatter/JavaScriptAssetUrlFormatter.php:111]: Seems like according to this article https://developer.mozilla.org/ru/docs/Web/HTML/Element/script it is redundant in HTML5 use `type` attribute. Drupal core for instance doing exposing javascript files without type attribute as well.

File: modules/oe_media_js_asset/tests/src/Kernel/JavaScriptAssetUrlFormatterTest.php [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+namespace Drupal\Tests\oe_media_js_asset\Kernel;
+
+use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\Tests\oe_media\Traits\MediaTypeCreationTrait;
+
+/**
+ * Tests the field JS asset url formatter.
```
Comment (sergepavle) [inline modules/oe_media_js_asset/tests/src/Kernel/JavaScriptAssetUrlFormatterTest.php:13]:
```suggestion
 * Tests the field JS asset URL formatter.
```

**Review by sergepavle — CHANGES_REQUESTED** [review]
Few very minor comments.
Didn't find anything in scope of EPIC regarding browser caching. Drupal core originally adds query arguments for possibility control browser caching with using `$query_string = $this->state->get('system.css_js_query_string', '0');`
value.
Is it reasonable to add support of this capability? Is it should be separate ticket?

**Review by sergepavle — APPROVED** [review]

**Pull request merged**